src/herder/HerderImpl.cpp

```diff
@@ -1934,9 +1950,69 @@ HerderImpl::restoreUpgrades()
     }
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+void
+HerderImpl::maybeHandleUpgrade()
+{
+    uint64_t diff = 0;
+    {
+        LedgerTxn ltx(mApp.getLedgerTxnRoot(),
+                      /* shouldUpdateLastModified */ true,
+                      TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+        auto conf = mApp.getLedgerManager().getSorobanNetworkConfig(ltx);
+
+        if (conf.txMaxSizeBytes() > mMaxTxSize)
+        {
+            diff = conf.txMaxSizeBytes() - mMaxTxSize;
+        }
+        mMaxTxSize =
```

Comment:
do we need to update FlowControl.cpp L388

Reply:
Are you referring to this? https://github.com/stellar/stellar-core/pull/3806/files#diff-d37d5bbb43097772a9cbca1b1215a7599e1fdd7be78ffb575fdc309dc9cf93d6R420 This PR already updates all places where getMaxTxSize should be used.